### ISSUE TITLE
list arguments break the code generation

### ISSUE THREAD

**Issue by parsodyl** [rrousselGit/freezed#25]
If you use the code from the example with generics and try to change the default factory putting a list as a parameter, you will encounter an error during the code generation:

```Dart
part 'union.freezed.dart';

@immutable
abstract class Union<T> with _$Union<T> {
  const factory Union(List<T> value) = Data<T>;
  const factory Union.apiError(WebException webException) = ApiErrorDetails<T>;
  const factory Union.connectionError() = ConnectionErrorDetails<T>;
}
```

```
Error formatting generated source code for [...] which was output to [...].
This may indicate an issue in the generated code or in the formatter.
Please check the generated code and file an issue on source_gen if appropriate.
Could not format because the source could not be parsed:

line 24, column 1: Expected to find ')'.
   ╷
24 │ }
   │ ^
   ╵
line 24, column 1: Expected to find ')'.
   ╷
24 │ }
   │ ^
   ╵
line 45, column 1: Expected to find ')'.
   ╷
45 │ }
   │ ^
   ╵
line 57, column 1: Expected to find ')'.
   ╷
57 │ }
   │ ^
   ╵
line 91, column 1: Expected to find ')'.
   ╷
91 │ }
   │ ^
   ╵
line 91, column 1: Expected to find ')'.
   ╷
91 │ }
   │ ^
   ╵
line 91, column 1: Expected to find ')'.
   ╷
91 │ }
   │ ^
   ╵
line 91, column 1: Expected to find ')'.
   ╷
91 │ }
   │ ^
   ╵
line 104, column 1: Expected to find ')'.
    ╷
104 │ }
    │ ^
    ╵
line 104, column 1: Expected to find ')'.
    ╷
104 │ }
    │ ^
    ╵
(8 more errors...)
[INFO] Running build completed, took 749ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 34ms

[SEVERE] Failed after 790ms

```

**Comment by rrousselGit** [comment]
Thanks for the report. I will fix it asap

**Comment by rrousselGit** [comment]
That's fixed as part of 0.2.3
